src/widgets/SettingsIcon.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { selectUsingVaccines } from '../selectors/modules';
+import { PermissionSelectors } from '../selectors/permission';
+import { PermissionActions } from '../actions/PermissionActions';
+
+import { useToggle } from '../hooks/index';
+import { CancelIcon, CogIcon, ConfirmIcon, DisabledIcon, HazardIcon } from './icons';
+import { PaperModalContainer } from './PaperModal/PaperModalContainer';
+import { IconButton } from './IconButton';
+import { FlexRow } from './FlexRow';
+
+import { DARK_GREY } from '../globalStyles';
+import { modalStrings } from '../localization';
+
+const PermissionRow = ({ comment, enabled, onPress, label }) => (
+  <TouchableOpacity onPress={enabled ? null : onPress}>
+    <FlexRow alignItems="flex-start" flex={0} style={localStyles.permissionRowContainer}>
+      {enabled ? <ConfirmIcon /> : <DisabledIcon />}
+      <FlexRow style={{ flexDirection: 'column', marginLeft: 10 }}>
+        <Text style={localStyles.permissionRowText}>{label}</Text>
+        <Text style={localStyles.permissionRowComment}>{comment}</Text>
+      </FlexRow>
+    </FlexRow>
+  </TouchableOpacity>
+);
+
+PermissionRow.defaultProps = {
+  comment: '',
+};
+
+PermissionRow.propTypes = {
+  comment: PropTypes.string,
+  enabled: PropTypes.bool.isRequired,
+  label: PropTypes.string.isRequired,
+  onPress: PropTypes.func.isRequired,
+};
+
+const SettingsIconComponent = ({
+  bluetooth,
+  location,
+  requestBluetooth,
+  requestLocation,
+  requestWriteStorage,
+  usingVaccines,
+  writeStorage,
+}) => {
+  const [settingsModalOpen, toggleSettingsModal] = useToggle();
+
+  if (!usingVaccines) {
+    return null;
+  }
+
+  const icon =
+    location && bluetooth && writeStorage ? (
+      <CogIcon color={DARK_GREY} />
+    ) : (
+      <HazardIcon color={DARK_GREY} size={20} />
+    );
+
+  return (
+    <>
+      <IconButton Icon={icon} containerStyle={{ marginLeft: 8 }} onPress={toggleSettingsModal} />
```

Comment:
I have fat fingers and had a litttle bit of an issue clicking this - not sure if it's worth the time to cater for this though 🤣